Revert WCPay international support for bundled package

joshuatf opened the pull request:
Reverts the international support for WCPay due to incomplete agreements.

Relevenat thread - paJDYF-15E-p2

cc @pmcpinto to confirm that this is the only flag we need to turn off.

### Detailed test instructions:

1. Build the core version of this branch.
2. Set your store to one of the following countries: `'AU', 'CA', 'GB', 'IE', 'NZ'`.
3. Make sure WCPay is not shown as a recommendation in the task list or payments task.

jeffstieler reviewed (approved):
Fix the changelog lint and :shipit:

joshuatf committed:
Revert WCPay international support for bundled package

joshuatf committed:
Add changelog entry

joshuatf commented:
Thanks, @jeffstieler!  Added a changelog entry and updated the base on this one to `2.2.2` since I think it's likely this will get included in the next bundled version of WCA in core.

joshuatf committed:
Merge branch 'release/2.2.2-rc.1' into revert/wcpay-international-support

timmyc commented:
Cherry picked to `release\2.3.0` in  36cc53d0870ce26c375e3609296c68f1d893bae4
